warnings from msvc 2022

just sharing its warnings but i have no time to take a look at them and submit a pr, 

```
cl : Command line warning D9025 : overriding '/MDd' with '/MT'
cl : Command line warning D9002 : ignoring unknown option '-fvisibility=hidden'
../shlr/qjs/src/quickjs.c(1351): warning C4244: 'function': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(6149): warning C4244: '+=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6150): warning C4244: '+=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6278): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6279): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6281): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6288): warning C4244: '+=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6464): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/quickjs.c(6497): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/quickjs.c(10328): warning C4244: '=': conversion from 'uint64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(11542): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(11557): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(40184): error C2099: initializer is not a constant
../shlr/qjs/src/quickjs.c(41005): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(46826): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(49156): error C2099: initializer is not a constant
../shlr/qjs/src/quickjs.c(49267): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(49294): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49295): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49296): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49297): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49298): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49299): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49300): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49301): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49302): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(49332): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49333): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49456): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49457): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49458): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49459): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49460): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49461): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49462): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49463): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49464): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(49563): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(51972): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(51978): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(52011): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(52016): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
[1977/2318] Compiling C object libr/lang/r_lang.dll.p/.._.._shlr_qjs_src_libbf.c.obj
cl : Command line warning D9025 : overriding '/MDd' with '/MT'
cl : Command line warning D9002 : ignoring unknown option '-fvisibility=hidden'
../shlr/qjs/src/libbf.c(484): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(1224): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(1233): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2403): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2404): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2405): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2575): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2591): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2627): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2652): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(3348): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(3391): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(4448): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(4981): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(7150): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(7168): warning C4146: unary minus operator applied to unsigned type, result still unsigned
ninja: build stopped: subcommand failed.
Ninja compilation has failed
```

thats what i get with latest commit:

```
l : Command line warning D9002 : ignoring unknown option '-fvisibility=hidden'
../shlr/qjs/src/quickjs.c(1381): warning C4244: 'function': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(6183): warning C4244: '+=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6184): warning C4244: '+=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6312): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6313): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6315): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6322): warning C4244: '+=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(6498): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/quickjs.c(6531): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/quickjs.c(10377): warning C4244: '=': conversion from 'uint64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(11591): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(11606): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(41109): error C2099: initializer is not a constant
../shlr/qjs/src/quickjs.c(41930): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(47856): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(50186): error C2099: initializer is not a constant
../shlr/qjs/src/quickjs.c(50297): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
../shlr/qjs/src/quickjs.c(50324): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50325): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50326): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50327): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50328): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50329): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50330): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50331): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50332): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(50362): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50363): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50486): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50487): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50488): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50489): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50490): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50491): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50492): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50493): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50494): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(50593): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(53174): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(53180): warning C4244: '=': conversion from 'double' to 'int', possible loss of data
../shlr/qjs/src/quickjs.c(53227): warning C4244: '=': conversion from 'int64_t' to 'double', possible loss of data
../shlr/qjs/src/quickjs.c(53232): warning C4244: '=': conversion from 'double' to 'int64_t', possible loss of data
[1977/2318] Compiling C object libr/lang/r_lang.dll.p/.._.._shlr_qjs_src_libbf.c.obj
cl : Command line warning D9025 : overriding '/MDd' with '/MT'
cl : Command line warning D9002 : ignoring unknown option '-fvisibility=hidden'
../shlr/qjs/src/libbf.c(484): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(1224): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(1233): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2403): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2404): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2405): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2575): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2591): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2627): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(2652): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(3348): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(3391): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(4448): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(4981): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(7150): warning C4146: unary minus operator applied to unsigned type, result still unsigned
../shlr/qjs/src/libbf.c(7168): warning C4146: unary minus operator applied to unsigned type, result still unsigned
ninja: build stopped: subcommand failed.
```

> ../shlr/qjs/src/quickjs.c(41109): error C2099: initializer is not a constant

That's the JS_ALIAS_BASE_DEF macro. I don't understand why msvc has trouble with it but we can probably get rid of it.

> ../shlr/qjs/src/quickjs.c(50186): error C2099: initializer is not a constant

That line number doesn't correspond to anything (blank line); what commit specifically are you testing?

my main concernn are the errors xD:

`../shlr/qjs/src/quickjs.c(41109): error C2099: initializer is not a constant`

and

`../shlr/qjs/src/quickjs.c(50186): error C2099: initializer is not a constant`

> @richarddd your report is on MinGW right? @trufae ia on MSVC.
> 
> At any rate, are the rust bindings using the CMake build system? If not, what compiler flags are you using?

No we use MSVC. I'm a total Windows novice so we might have done something wrong. We use CC crate withthese build flags: `/std:c11 /experimental:c11atomics`
https://github.com/DelSkayn/rquickjs/blob/master/sys/build.rs#L159

This is the command that failes:
```shell
quickjs.c
  D:\a\llrt\llrt\target\debug\build\rquickjs-sys-683cbfbb52ef3647\out\quickjs.c(40539): error C2099: initializer is not a constant
  D:\a\llrt\llrt\target\debug\build\rquickjs-sys-683cbfbb52ef3647\out\quickjs.c(49511): error C2099: initializer is not a constant
  exit code: 0
  exit code: 0
  exit code: 0
  exit code: 2
  libbf.c
  cargo:warning=ToolExecError: Command "C:\\Program Files\\Microsoft Visual Studio\\2022\\Enterprise\\VC\\Tools\\MSVC\\14.42.34433\\bin\\HostX64\\x64\\cl.exe" "-nologo" "-MD" "-Brepro" "/std:c11" "/experimental:c11atomics" "-D_GNU_SOURCE" "-FoD:\\a\\llrt\\llrt\\target\\debug\\build\\rquickjs-sys-683cbfbb52ef3647\\out\\862492ac8e08571a-quickjs.o" "-c" "D:\\a\\llrt\\llrt\\target\\debug\\build\\rquickjs-sys-683cbfbb52ef3647\\out\\quickjs.c" with args cl.exe did not execute successfully (status code exit code: 2).

```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -308,6 +308,36 @@ jobs:
           build\run-test262.exe -c tests.conf
           build\function_source.exe
 
+  windows-sdk:
+    runs-on: windows-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        arch: [x64, Win32]
+        buildType: [Debug, Release]
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install windows sdk
+        uses: ChristopheLav/windows-sdk-install@v1
+        with:
+          version-sdk: 26100
+          features: 'OptionId.DesktopCPPx86,OptionId.DesktopCPPx64'
+      - name: build
+        run: |
+          cmake -B build -DBUILD_EXAMPLES=ON -DCMAKE_SYSTEM_VERSION="10.0.26100.0" -A ${{matrix.arch}}
+          cmake --build build --config ${{matrix.buildType}}
+      - name: stats
+        run: |
+          build\${{matrix.buildType}}\qjs.exe -qd
+      - name: test
+        run: |
+          cp build\${{matrix.buildType}}\fib.dll examples\
+          cp build\${{matrix.buildType}}\point.dll examples\
+          build\${{matrix.buildType}}\qjs.exe examples\test_fib.js
+          build\${{matrix.buildType}}\qjs.exe examples\test_point.js
+          build\${{matrix.buildType}}\run-test262.exe -c tests.conf
+          build\${{matrix.buildType}}\function_source.exe
+
   windows-mingw:
     runs-on: windows-latest
     strategy:

--- a/quickjs.c
+++ b/quickjs.c
@@ -41116,7 +41116,7 @@ static const JSCFunctionListEntry js_number_funcs[] = {
     JS_CFUNC_DEF("isSafeInteger", 1, js_number_isSafeInteger ),
     JS_PROP_DOUBLE_DEF("MAX_VALUE", 1.7976931348623157e+308, 0 ),
     JS_PROP_DOUBLE_DEF("MIN_VALUE", 5e-324, 0 ),
-    JS_PROP_DOUBLE_DEF("NaN", NAN, 0 ),
+    JS_PROP_U2D_DEF("NaN", 0x7FF8ull<<48, 0 ), // workaround for msvc
     JS_PROP_DOUBLE_DEF("NEGATIVE_INFINITY", -INFINITY, 0 ),
     JS_PROP_DOUBLE_DEF("POSITIVE_INFINITY", INFINITY, 0 ),
     JS_PROP_DOUBLE_DEF("EPSILON", 2.220446049250313e-16, 0 ), /* ES6 */
@@ -50197,7 +50197,7 @@ static const JSCFunctionListEntry js_global_funcs[] = {
     JS_CFUNC_DEF("escape", 1, js_global_escape ),
     JS_CFUNC_DEF("unescape", 1, js_global_unescape ),
     JS_PROP_DOUBLE_DEF("Infinity", 1.0 / 0.0, 0 ),
-    JS_PROP_DOUBLE_DEF("NaN", NAN, 0 ),
+    JS_PROP_U2D_DEF("NaN", 0x7FF8ull<<48, 0 ), // workaround for msvc
     JS_PROP_UNDEFINED_DEF("undefined", 0 ),
     JS_PROP_STRING_DEF("[Symbol.toStringTag]", "global", JS_PROP_CONFIGURABLE ),
 };

--- a/quickjs.h
+++ b/quickjs.h
@@ -970,6 +970,7 @@ typedef struct JSCFunctionListEntry {
         const char *str;    /* pure ASCII or UTF-8 encoded */
         int32_t i32;
         int64_t i64;
+        uint64_t u64;
         double f64;
     } u;
 } JSCFunctionListEntry;
@@ -998,6 +999,7 @@ typedef struct JSCFunctionListEntry {
 #define JS_PROP_INT32_DEF(name, val, prop_flags) { name, prop_flags, JS_DEF_PROP_INT32, 0, { .i32 = val } }
 #define JS_PROP_INT64_DEF(name, val, prop_flags) { name, prop_flags, JS_DEF_PROP_INT64, 0, { .i64 = val } }
 #define JS_PROP_DOUBLE_DEF(name, val, prop_flags) { name, prop_flags, JS_DEF_PROP_DOUBLE, 0, { .f64 = val } }
+#define JS_PROP_U2D_DEF(name, val, prop_flags) { name, prop_flags, JS_DEF_PROP_DOUBLE, 0, { .u64 = val } }
 #define JS_PROP_UNDEFINED_DEF(name, prop_flags) { name, prop_flags, JS_DEF_PROP_UNDEFINED, 0, { .i32 = 0 } }
 #define JS_OBJECT_DEF(name, tab, len, prop_flags) { name, prop_flags, JS_DEF_OBJECT, 0, { .prop_list = { tab, len } } }
 #define JS_ALIAS_DEF(name, from) { name, JS_PROP_WRITABLE | JS_PROP_CONFIGURABLE, JS_DEF_ALIAS, 0, { .alias = { from, -1 } } }

